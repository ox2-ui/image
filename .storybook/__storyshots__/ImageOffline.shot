exports[`broken`] = `
<img
  className=""
  onError={[Function]}
  src="/images/BROKENmGxer98WQqTudkdWK.jpg"
  style={undefined} />
`;

exports[`broken offline`] = `
<img
  className=""
  onError={[Function]}
  src="/images/BROKENmGxer98WQqTudkdWK.jpg"
  style={undefined} />
`;

exports[`offline`] = `
<img
  className=""
  onError={[Function]}
  src="/images/mGxer98WQqTudkdWK.jpg"
  style={undefined} />
`;
