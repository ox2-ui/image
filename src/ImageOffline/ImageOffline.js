import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

/**
 * ImageOffline Component
 */
class ImageOffline extends PureComponent {
  static propTypes = {
    /**
      * The css class name of the root element.
      */
    className: PropTypes.string,
    /**
      * Offline/local server image filename
      */
    localImg: PropTypes.string.isRequired,
    /**
      * Image remote server url
      */
    remoteLink: PropTypes.string.isRequired,
    /**
      * Override the inline-styles of the root element.
      */
    style: PropTypes.object,
  };

  static defaultProps = {
    className: '',
    style: {},
  };

  state = {
    loadOffline: true,
  };

  handleError = () => {
    this.setState({ loadOffline: false });
  };

  render() {
    const {
      className,
      remoteLink,
      localImg,
      style,
    } = this.props;
    const { loadOffline } = this.state;

    const offlineLink = `/images/${localImg}`;

    let img;
    if (loadOffline) {
      img = (
        <img
          className={className}
          onError={this.handleError}
          src={offlineLink}
          style={style}
        />
      );
    } else {
      img = (
        <img
          className={className}
          src={remoteLink}
          style={style}
        />
      );
    }

    return img;
  }
}

export default ImageOffline;
