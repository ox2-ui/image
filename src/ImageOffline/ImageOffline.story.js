import React from 'react';
import { storiesOf } from '@storybook/react';
import ImageOffline from './ImageOffline';

const remoteLink = 'http://imgur.com/h0UFha7.jpg';

storiesOf('ImageOffline', module)
  .add('offline', () =>
    <ImageOffline
      localImg={'mGxer98WQqTudkdWK.jpg'}
      remoteLink={remoteLink}
    />,
  )
  .add('broken offline', () =>
    <ImageOffline
      localImg={'BROKENmGxer98WQqTudkdWK.jpg'}
      remoteLink={remoteLink}
    />,
  )
  .add('broken', () =>
    <ImageOffline
      localImg={'BROKENmGxer98WQqTudkdWK.jpg'}
      remoteLink={'broken'}
    />,
  );
