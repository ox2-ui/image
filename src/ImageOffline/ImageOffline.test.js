/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ImageOffline from './ImageOffline';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <ImageOffline
      localImg={'mGxer98WQqTudkdWK.jpg'}
      remoteLink={'http://imgur.com/h0UFha7.jpg'}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
