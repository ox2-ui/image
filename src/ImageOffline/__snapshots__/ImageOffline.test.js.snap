// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ImageOffline
  className=""
  localImg="mGxer98WQqTudkdWK.jpg"
  remoteLink="http://imgur.com/h0UFha7.jpg"
  style={Object {}}
>
  <img
    className=""
    onError={[Function]}
    src="/images/mGxer98WQqTudkdWK.jpg"
    style={Object {}}
  />
</ImageOffline>
`;
